RUFAS/graph_generator.py

```diff
@@ -0,0 +1,42 @@
+import matplotlib
+
+matplotlib.use('TkAgg')
+import matplotlib.pyplot as plt
+
+
+class GraphGenerator:
+    def generate_graph(self, filtered_pool: dict, graph_info: dict, graph_path: str, **kwargs):
+        graph_type = graph_info["graph_type"]
+        if graph_type == 'Line graph':
+            self._line_graph(filtered_pool, graph_info, graph_path)
+        elif graph_type == 'Bar graph':
+            self._bar_graph(filtered_pool, graph_info, graph_path)
```

Comment:
This structure avoids code duplication and makes it easier down the road to add new graph types.
```suggestion
        graph_type_funcs={
        "Line Graph": self._line_graph,
        "Bar Graph": self._bar_graph,
        }
        graph_type_funcs[graph_info["graph_type"]](filtered_pool, graph_info, graph_path)
```